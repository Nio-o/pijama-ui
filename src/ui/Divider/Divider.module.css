.Divider {
  border-width: var(--qw-divider-border-width);
  border-style: var(--qw-divider-border-style);
  border-color: var(--qw-divider-border-color);
}
