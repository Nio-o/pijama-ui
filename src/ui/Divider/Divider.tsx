import type { ComponentType, ElementType, ReactElement } from 'react'
import React from 'react'

import { cn } from '@bem-react/classname'

import s from './Divider.module.js'

export type DividerElementType = ElementType | ComponentType

export interface DividerProps<T extends DividerElementType> {
  as?: T
  className?: string
  testId?: string
}

const cnDivider = cn(s.Divider)

export const Divider = <T extends DividerElementType = 'div'>({
  as,
  className,
  testId,
}: DividerProps<T>): ReactElement | null => {
  const Component = as || 'div'
  return <Component data-testid={testId} role="separator" className={cnDivider(null, [className])} />
}
