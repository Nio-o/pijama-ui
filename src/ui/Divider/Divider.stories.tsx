import React from 'react'

import type { ComponentMeta, ComponentStory } from '@storybook/react'

import { Divider } from 'pijama/ui/Divider/bundle'

export default {
  title: 'Components/Divider',
  component: Divider,
} as ComponentMeta<typeof Divider>

export const Playground: ComponentStory<typeof Divider> = () => {
  return (
    <>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Eius animi est doloremque, ullam dolorum eum
        consequuntur ea nulla cumque quo?
      </p>
      <Divider />
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Recusandae id molestias saepe obcaecati eius delectus
        enim, perspiciatis voluptatibus, rerum odit eveniet aut! Hic laboriosam nulla, accusantium itaque nemo ratione
        inventore.
      </p>
      <Divider />
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Recusandae id molestias saepe obcaecati eius delectus
        enim, perspiciatis voluptatibus, rerum odit eveniet aut! Hic laboriosam nulla, accusantium itaque nemo ratione
        inventore.
      </p>
    </>
  )
}
