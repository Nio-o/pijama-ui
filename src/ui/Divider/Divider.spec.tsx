import React from 'react'

import { render } from '@testing-library/react'

import { Divider } from './Divider'

import '@testing-library/jest-dom'

describe('Components / Divider', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Divider />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Should have role="separator"', () => {
    const { container } = render(<Divider />)
    expect(container.firstChild).toHaveAttribute('role', 'separator')
  })

  it('Should render as "div" by default', () => {
    const container = render(<Divider testId="test" />)
    expect(container.getByTestId('test').tagName.toLowerCase()).toBe('div')
  })

  it('Should render as "span" when prop "as=span" provided', () => {
    const container = render(<Divider testId="test" as="span" />)
    expect(container.getByTestId('test').tagName.toLowerCase()).toBe('span')
  })
})
