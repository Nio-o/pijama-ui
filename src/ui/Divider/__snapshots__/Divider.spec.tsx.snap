// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Divider Should match snapshot 1`] = `
<div
  class="Divider"
  role="separator"
/>
`;
